src/contexts/ToolOptionsContext.tsx

```diff
@@ -31,4 +31,14 @@ export const useToolOptions = () => {
   return context
 }
 
+export const withToolOptions =
```

Comment:
An alternative approach to this, is to use `studio.components.layout` config in `sanity.config.ts`, and put your Context provider there. That way, everything in the studio has access, regardless of location, and you dont have to juggle all these `withThoolOptions` 

Up to you.

```
// under sanity config
studio: {
  components: {
    layout: (props) => {
      return <ToolOptionsProvider>{props.renderDefault(props)}</ToolOptionsProvider>
    },
  },
},
```

Reply:
Oh, that is smooth! I like it 💯